pkgs/development/python-modules/pathy/default.nix

```
@@ -0,0 +1,35 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, typer
+, dataclasses
+, smart_open
+, pytest
+, mock
+, google-cloud-storage
+}:
+
+buildPythonPackage rec {
+  pname = "pathy";
+  version = "0.5.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-nb8my/5rkc7thuHnXZHe1Hg8j+sLBlYyJcLHWrrKZ5M=";
+  };
+
+  propagatedBuildInputs = [ smart_open typer google-cloud-storage ];
+
+  postPatch = ''
+    substituteInPlace requirements.txt --replace "smart-open>=2.2.0,<4.0.0" "smart-open>=2.2.0"
```

just a little more readable
```suggestion
    substituteInPlace requirements.txt \
      --replace "smart-open>=2.2.0,<4.0.0" "smart-open>=2.2.0"
```